src/disco/topo/fd_topo.c

```diff
@@ -318,14 +314,14 @@ fd_topo_workspace_fill( fd_topo_t *      topo,
       fd_topo_tile_t * tile = &topo->tiles[ i ];
       if( FD_LIKELY( tile->wksp_id!=wksp->id ) ) continue;
 
-      fd_tile_config_t * config = fd_topo_tile_to_config( tile );
-      if( FD_LIKELY( config->scratch_align ) ) {
+      ulong align = tile_align( tile );
+      if( FD_LIKELY( align ) ) {
```

Comment:
Oh now I see this align of 0 means no memory gets allocated. Are there any tiles with no footprint? I think if align is 1 and footprint is 0, you can get rid of this branch and it won't allocate any space for the tile.